### PR TITLE
 Move lombok dependency into root dependency manager for easier version management

### PR DESCRIPTION
**Pull request by JohnNiang**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [x] New feature provided
- [ ] Improve performance

- Related issues

https://github.com/apache/skywalking/issues/4284

___
### New feature or improvement

Move lombok dependency into root dependency manager for easier version management.

### CODE REVIEW

**Comment by JohnNiang** [comment]
Maybe I have to downgrade lombok version.

**Comment by kezhenxu94** [comment]
> Maybe I have to downgrade lombok version.

@JohnNiang no, using the latest Lombok version should be ok

File: pom.xml [hunk 1]
```diff
@@ -275,6 +276,12 @@
                 <version>${powermock.version}</version>
                 <scope>test</scope>
             </dependency>
+            <dependency>
```
Comment (JohnNiang) [inline pom.xml]: @kezhenxu94 Should I move the `lombok` into root `dependencies` other than `dependencyManagement`.

Comment (wu-sheng) [reply pom.xml]: That depends on how you use it. Logically, they are the same. But `provided` is pointless.

Comment (wu-sheng) [reply pom.xml]: And from the current `dependencies` in the root `pom.xml`, no lib is declared there.

**Comment by wu-sheng** [comment]
You should test the compiling locally first.

**Comment by JohnNiang** [comment]
> You should test the compiling locally first.

The checks of test folder failed due to redundant dependency.

File: pom.xml (hunk 1 above)
Comment (kezhenxu94) [reply pom.xml]: It should be better, and remove all Lombok in all sub modules

**Pull request merged**